### PR TITLE
SteeringMotor{} rewrite

### PR DESCRIPTION
**Pull request by AzulRadio**
- Rewrite `AlignUpdate()`
- Improve `TurnRelative()` `GetTheta()` implementation
- Change a few function names
- Remove / Add some private members
- Change downstream libraries accordingly
- Add example for 2 motors alignment (calibration) in `m3508_steering.cc`

### CODE REVIEW

File: examples/motor/m3508_steering.cc [hunk 1]
```diff
@@ -51,63 +54,89 @@ bool steering_align_detect() {
 
 void RM_RTOS_Init() {
   print_use_uart(&huart8);
-  bsp::SetHighresClockTimer(&htim2);
```
Comment (RogerQi) [inline examples/motor/m3508_steering.cc:54]: Does this only work for typeA board?

File: examples/motor/m3508_steering.cc [hunk 2]
```diff
@@ -51,63 +54,89 @@ bool steering_align_detect() {
...
+  bsp::SetHighresClockTimer(&htim5);
+
+  can1 = new bsp::CAN(&hcan1, 0x201, true);
+  motor1 = new control::Motor3508(can1, 0x201);
+  motor3 = new control::Motor3508(can1, 0x203);
 
-  can1 = new bsp::CAN(&hcan1, 0x201);
-  motor = new control::Motor3508(can1, 0x202);
+  key = new bsp::GPIO(KEY_GPIO_GROUP, KEY_GPIO_PIN);
```
Comment (RogerQi) [inline examples/motor/m3508_steering.cc:63]: Add a comment here - what is the purpose of key? I am assuming that key on the dev board is pressed to trigger the alignment process

**Review by RogerQi — APPROVED** [review]
libraries code looks well-documented. Some comments are needed to explain things in example/motor/m3508_steering.cc

File: examples/motor/m3508_steering.cc (hunk 1 above)
Comment (AzulRadio) [reply examples/motor/m3508_steering.cc:54]: Yes. The example is written against typeA and not tested on typeC.

File: examples/motor/m3508_steering.cc (hunk 2 above)
Comment (AzulRadio) [reply examples/motor/m3508_steering.cc:68]: Resolved in 3dea93c and b145258

**Pull request merged**